### ISSUE TITLE
Nomad ACL Policy for variables can't accessed when defined with leading slash

### ISSUE THREAD

**Issue by rahadiangg** [hashicorp/nomad#23730]
### Nomad version
Nomad v1.8.2
BuildDate 2024-07-16T08:50:09Z
Revision 7f0822c1e4f25907d9f60e2d595411950dd1bd28

### Operating system and Environment details
Ubuntu 22.04
AWS t3.micro

### Issue
When definied acl policy with root trailing slash like below, the nomad task got 403 permission denied

```
namespace "default" {
  variables {
    path "/secret/aws/*" {
      capabilities = [
        "read",
        "list"
      ]
    }
  }
}
```

but when we remove the root trailing slash, the nomad task run as expedted

```
path "secret/aws/*"
```

### Reproduction steps
Assign policy that allow access to specified variables path with root trailing slash. E.g `nomad acl policy apply -namespace=default -job="2.access-variables-non-standard" allow-access-aws-secret.hcl`

#### Expected Result
Task running with rendered file from variables data

#### Actual Result
Pending task

### Job file (if appropriate)

<details>

```
job "2.access-variables-non-standard" {

  namespace = "default"

  group "dummy-group" {

    task "dummy-task" {
      driver = "exec"

      // driver configuration
      config {
        command = "/bin/bash"
        args = [
          "-c", "echo 'Task 1 is running'; sleep 3600"
        ]
      }

      resources {
        cpu    = 100 # MHz
        memory = 100 # MB
      }

      template {
        env         = false
        destination = "local/aws-config-demo"
        change_mode = "restart"
        data        = <<EOF
{{- with nomadVar "secret/aws/demo-account" }}
{{- range $key, $value := . }}
{{ $key }}={{ $value }}
{{- end }}
{{- end }}
EOF
      }

      template {
        env         = false
        destination = "local/aws-config-staging"
        change_mode = "restart"
        data        = <<EOF
{{- with nomadVar "secret/aws/staging-account" }}
{{- range $key, $value := . }}
{{ $key }}={{ $value }}
{{- end }}
{{- end }}
EOF
      }
    }
  }
}
```

</details>

### Nomad Server logs (if appropriate)

```
Aug 04 05:03:20 ip-172-31-0-241 nomad[5751]:     2024-08-04T05:03:20.493Z [DEBUG] http: request complete: method=GET path="/v1/var/secret/aws/staging-account?index=1&namespace=default&stale=&wait=300000ms" duration="205.364µs"
Aug 04 05:03:20 ip-172-31-0-241 nomad[5751]:     2024-08-04T05:03:20.571Z [DEBUG] http: request failed: method=GET path="/v1/var/secret/aws/demo-account?index=1&namespace=default&stale=&wait=300000ms" error="Permission denied" code=403
```

**Comment by tgross** [comment]
Hi @rahadiangg! As you saw, the ACL path has to match the Variable's path, and there is no "root" for Variables. That's why your `template` block has `{{- with nomadVar "secret/aws/staging-account" }}` without the leading `/`.

However, we do something misleading with the API -- we strip off the leading `/` when writing or reading a Variable. Which can make you think that's the correct canonical path unless you're paying very close attention. Notice in the responses below that the leading slash is gone.

```
$ nomad var put /example/foo name=Tim
Created variable "example/foo" with modify index 24

$ nomad var get /example/foo
Namespace   = default
Path        = example/foo
Create Time = 2024-08-05T09:16:40-04:00
Check Index = 24

Items
name = Tim
```

That's a nice convenience but ends up being a bad UX when we compare those paths to ACL policy paths. I'm somewhat hesitant to change the ACL policy behavior here. If we were to change it I'd most likely want to make the ACL policy you submitted return an error; for ACLs is probably better to force an error than silently change the value. Let me check in with the rest of my team to see if we have consensus on that, and then I'll circle back here.

**Comment by rahadiangg** [comment]
Hi @tgross, thank for your answer and explaination. I think that also happend in Nomad web UI

![ScreenRecording2024-08-06at10 47 42-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/e0b92cbd-f1fb-44a4-ac42-813accd5b27a)

**Comment by tgross** [comment]
#23757 has been merged and will ship in Nomad 1.8.3
